ui/src/testing/stubs/workspaces-api-stub.ts

```diff
@@ -1,14 +1,35 @@
 import {
-  FileDetail,
+  FileDetail, ShareWorkspaceRequest, ShareWorkspaceResponse,
+  UserRole,
+  Workspace,
+  WorkspaceAccessLevel,
+  WorkspaceResponseListResponse,
   WorkspacesApi
 } from 'generated/fetch';
 
+import {UserServiceStub} from './user-service-stub';
+
+import * as fp from 'lodash/fp';
+
+import {EmptyResponse} from 'generated';
+
+export class WorkspaceStubVariables {
+  static DEFAULT_WORKSPACE_NS = 'defaultNamespace';
+  static DEFAULT_WORKSPACE_NAME = 'defaultWorkspace';
+  static DEFAULT_WORKSPACE_ID = '1';
+  static DEFAULT_WORKSPACE_DESCRIPTION = 'Stub workspace';
+  static DEFAULT_WORKSPACE_CDR_VERSION = 'Fake CDR Version';
+}
+
 export class WorkspacesApiStub extends WorkspacesApi {
+  workspaces: Workspace[];
+  workspaceAccess: Map<string, WorkspaceAccessLevel>;
   notebookList: FileDetail[];
 
-  constructor() {
+  constructor(workspaces?: Workspace[]) {
     super(undefined, undefined, (..._: any[]) => { throw Error('cannot fetch in tests'); });
-
+    this.workspaces = fp.defaults([WorkspacesApiStub.stubWorkspace()], workspaces);
```

Comment:
I don't think `defaults` is the right fit here - looks like it's intended purpose is to fill in missing property values; I'm not sure what the expected behavior in this case of arrays is, but `[]` should be a legal input and it's handling that incorrectly here.

Reply:
Hmmm, defaultTo then? I think that may be actually what I meant to use.